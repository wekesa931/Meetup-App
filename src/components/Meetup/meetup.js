import React from 'react';
import SingleMeetup from '../../containers/single_meetup_container';

const Meetup = (props) => {
    return (
        <div>
            <SingleMeetup {...props}/>
        </div>
    );
};

export default Meetup;