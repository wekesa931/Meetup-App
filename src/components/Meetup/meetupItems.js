import React from 'react';
import {Link} from 'react-router-dom';
import CreateMeetup from '../Create/addmeetup';
import DeleteContainer from '../../containers/delete_meetup_container';


const MeetupItems = (props) => {
    const renderMeetups = (meetups) => (
        meetups ? 
            meetups.map((item)=>(
                <div key={item.id}
                    style={{
                        boxShadow: 'rgb(136, 136, 136) 0px 0px 7px 0px',
                        padding: '3px',
                        marginBottom: '10px'
                    }}
                >
                    <Link to={`/meetups/${item.id}`}
                            style={{
                                textDecoration:'none',
                                color: 'black'
                            }}
                    >
                        <div><h3>{item.name}</h3></div>
                        <div>{item.title}</div> 
                        <div>Created On: {item.createdon}</div>
                    </Link>
                    <DeleteContainer id={item.id}/>
                </div>
            ))
        : null
    )
    return (
        <div>
            {renderMeetups(props.meetups)}
            <CreateMeetup/>
        </div>
    );
};

export default MeetupItems;