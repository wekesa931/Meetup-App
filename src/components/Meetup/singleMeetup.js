import React from 'react';

const DisplayMeetup = (props) => {
    console.log(props.mtp)
    const renderMeetup = (meetup) => (
        meetup ? meetup.map((item)=>(
            <div key={item.id}>
                <div><h3>{item.name}</h3></div>
                <div><strong>{item.title}</strong></div><br/>
                <div>{item.details}</div><br/>
                <div>Created on: {item.createdon}</div>
            </div>
        )):null
    )
    return (
        <div>
            {renderMeetup(props.mtp)}
        </div>
    );
};

export default DisplayMeetup;