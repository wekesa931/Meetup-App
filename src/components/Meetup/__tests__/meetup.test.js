import React from "react";
import Enzyme, { shallow } from "enzyme";
import Meetup from '../meetup';
import EnzymeAdapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe('Meetup', () => {
    const wrapper = shallow(<Meetup/>);
    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
})