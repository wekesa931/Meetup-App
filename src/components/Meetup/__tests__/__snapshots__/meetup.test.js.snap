// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meetup renders correctly 1`] = `ShallowWrapper {}`;
